core/src/main/java/org/arquillian/cube/impl/client/container/ContainerConfigurationController.java

```diff
@@ -64,12 +67,20 @@ public void remapContainer(@Observes BeforeSetup event, CubeRegistry cubeRegistr
                     int containerPort = getDefaultPortFromConfigurationInstance(newConfigurationInstance,
                             configurationClass, configurationClassPortField);
 
-                    PortBinding bindingForExposedPort = null;
-                    if ((bindingForExposedPort = binding.getBindingForExposedPort(containerPort)) != null) {
+                    PortBinding bindingForPort = null;
+                    if ((bindingForPort = binding.getBindingForExposedPort(containerPort)) != null) {
```

Comment:
Could you add a Metadata impl for DockerCube as well that rely on Cube.binding() for now. Then we can at least focus on only one of the 'methods of getting binding data' in this part of the code. Maybe add a @deprecated to Cube.bindings() method.